upcloud_api/cloud_manager/storage_mixin.py

```diff
@@ -108,3 +108,59 @@ def detach_storage(self, server, address):
         url = '/server/{0}/storage/detach'.format(server)
         res = self.post_request(url, body)
         return Storage._create_storage_objs(res['server']['storage_devices'], cloud_manager=self)
+
+    def load_cd_rom(self, server, address):
+        """
+        Loads a storage as a CD-ROM in the CD-ROM device of a server.
+        """
+        body = {'storage_device': {'storage': address}}
+        url = '/server/{0}/cdrom/load'.format(server)
+        res = self.post_request(url, body)
+        return Storage._create_storage_objs(res['server']['storage_devices'], cloud_manager=self)
+
+    def eject_cd_rom(self, server):
+        """
+        Ejects the storage from the CD-ROM device of a server.
+        """
+        url = '/server/{0}/cdrom/eject'.format(server)
+        res = self.post_request(url)
+        return Storage._create_storage_objs(res['server']['storage_devices'], cloud_manager=self)
+
+    def create_storage_backup(self, storage, title):
+        """
+        Creates a point-in-time backup of a storage resource.
+        """
+        url = '/storage/{0}/backup'.format(storage)
+        body = {'storage': {'title': title}}
+        res = self.post_request(url, body)
+        return Storage(cloud_manager=self, **res['storage'])
+
+    def restore_storage_backup(self, storage):
+        """
+        Restores the origin storage with data from the specified backup storage.
+        """
+        url = '/storage/{0}/restore'.format(storage)
+        return self.post_request(url)
+
+    def templatize_storage(self, storage, title):
+        """
+        Creates an exact copy of an existing storage resource which can be used as a template for creating new servers.
+        """
+        url = '/storage/{0}/templatize'.format(storage)
+        body = {'storage': {'title': title}}
+        res = self.post_request(url, body)
+        return Storage(cloud_manager=self, **res['storage'])
+
+    def add_storage_to_favorites(self, storage):
+        """
+        Adds a storage to the list of favorite storages.
+        """
+        url = '/storage/{0}/favorite'.format(storage)
+        return self.post_request(url)
+
+    def remove_storage_from_favorites(self, storage):
```

Comment:
Hmm, I wonder if anyone uses the favorite system that much anymore 🤔  The UpCloud Hub doesn't support it atleast 😄

Reply:
Hmm. yeah, maybe leave favorites out, since they might drop from support all together.